tizen/src/workmanager_tizen_plugin.cc

```diff
@@ -0,0 +1,347 @@
+#include "workmanager_tizen_plugin.h"
+
+#include <app_preference.h>
+#include <flutter/encodable_value.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar.h>
+#include <flutter/standard_method_codec.h>
+#include <job_scheduler.h>
+#include <system_info.h>
+#include <tizen.h>
+
+#include <memory>
+#include <string>
+
+#include "extractor.h"
+#include "job_scheduler_wrapper.h"
+#include "log.h"
+
+namespace {
+
+typedef flutter::MethodCall<flutter::EncodableValue> FlMethodCall;
+typedef flutter::MethodResult<flutter::EncodableValue> FlMethodResult;
+typedef flutter::MethodChannel<flutter::EncodableValue> FlMethodChannel;
+
+class WorkmanagerTizenPlugin : public flutter::Plugin {
+   public:
+    static void RegisterWithRegistrar(flutter::PluginRegistrar* registrar) {
+        auto plugin = std::make_unique<WorkmanagerTizenPlugin>();
+
+        auto foreground_channel =
+            std::make_unique<FlMethodChannel>(
+                registrar->messenger(), constants::kForegroundChannelName,
+                &flutter::StandardMethodCodec::GetInstance());
+
+        plugin->background_channel_ =
+            std::make_unique<FlMethodChannel>(
+                registrar->messenger(), constants::kBackgroundChannelName,
+                &flutter::StandardMethodCodec::GetInstance());
+
+        foreground_channel->SetMethodCallHandler(
+            [plugin_pointer = plugin.get()](const auto& call, auto result) {
+                plugin_pointer->HandleWorkmanagerCall(call, std::move(result));
+            });
+
+        plugin->background_channel_->SetMethodCallHandler(
+            [plugin_pointer = plugin.get()](const auto& call, auto result) {
+                plugin_pointer->HandleBackground(call, std::move(result));
+            });
+
+        registrar->AddPlugin(std::move(plugin));
+    }
+
+    WorkmanagerTizenPlugin() {}
+
+    virtual ~WorkmanagerTizenPlugin() {}
+
+   private:
+    std::unique_ptr<FlMethodChannel>
+        background_channel_;
+
+    void HandleWorkmanagerCall(const FlMethodCall& call,
```

Comment:
How about applying this [form](https://github.com/flutter-tizen/plugins/blob/c23e61c21935a9204fb7169b6892ec26ab1f1fd5/packages/flutter_tts/tizen/src/flutter_tts_tizen_plugin.cc#L77-L116).

Reply:
Fixed but I passed result as parameters.

Reply:
I integrated all foreground handlers in to one method so the form is violated now.